app/services/declarations/query.rb

```diff
@@ -0,0 +1,62 @@
+module Declarations
+  class Query
+    include Queries::ConditionFormats
+
+    attr_reader :scope, :sort
+
+    def initialize(lead_provider: :ignore, updated_since: :ignore, participant_ids: :ignore, cohort_start_years: :ignore)
+      @scope = all_declarations
+      @sort = sort
+
+      where_lead_provider_is(lead_provider)
+      where_updated_since(updated_since)
+      where_participant_ids_in(participant_ids)
+      where_cohort_start_year_in(cohort_start_years)
+    end
+
+    def declarations
+      scope.order(created_at: :asc)
+    end
+
+  private
+
+    def where_lead_provider_is(lead_provider)
+      return if lead_provider == :ignore
+
+      scope.merge!(Declaration.where(lead_provider:))
+    end
+
+    def where_updated_since(updated_since)
+      return if updated_since == :ignore
+
+      scope.merge!(Declaration.where(updated_at: updated_since..))
+    end
+
+    def where_participant_ids_in(participant_ids)
+      return if participant_ids == :ignore
+
+      scope.merge!(Declaration.where(user: { ecf_id: extract_conditions(participant_ids) }))
```

Comment:
existing issue in ECF: if you pass in a participant id that isn't in UUID format it returns a 500, one to tackle I guess in a different ticket

Reply:
Ah good catch; yeah I think pick that up as a separate ticket if its an issue throughout the app